static/app/views/onboarding/targetedOnboarding/components/stepper.tsx

```diff
@@ -0,0 +1,77 @@
+import {useEffect, useRef} from 'react';
+import styled from '@emotion/styled';
+import {animate, HTMLMotionProps, motion, useMotionValue} from 'framer-motion';
+
+const StepperWrapper = styled(motion.div)`
+  border-radius: 4px;
+  position: relative;
+  overflow: hidden;
+`;
+StepperWrapper.defaultProps = {
+  initial: {opacity: 0},
+  animate: {opacity: 1},
+  exit: {opacity: 0},
+};
+const StepperContainer = styled('div')`
+  display: flex;
+  flex-direction: row;
+  gap: 8px;
+`;
+
+const StepperIndicator = styled('span')<{clickable?: boolean}>`
+  height: 8px;
+  width: 80px;
+  background-color: ${p => p.theme.progressBackground};
+  cursor: ${p => (p.clickable ? 'pointer' : 'default')};
+`;
+
+const StepperTransitionIndicator = styled(motion.span)`
+  height: 8px;
+  width: 80px;
+  background-color: ${p => p.theme.progressBar};
+  position: absolute;
+`;
+
+type Props = HTMLMotionProps<'div'> & {
+  currentStepIndex: number;
+  numSteps: number;
+  onClick: (stepIndex: number) => void;
+};
+
+export default function Stepper({currentStepIndex, numSteps, onClick, ...props}: Props) {
+  const stepperContainerRef = useRef<HTMLDivElement>(null);
+  const stepperX = useMotionValue<null | number>(null);
+
+  // Set initial value of stepperX
+  useEffect(() => {
+    const parentRect = stepperContainerRef.current?.getBoundingClientRect();
+    const rect =
+      stepperContainerRef.current?.children[currentStepIndex].getBoundingClientRect();
+    if (!rect || !parentRect) {
+      return;
+    }
+    if (stepperX.get() === null) {
+      stepperX.set(rect.x - parentRect.x);
+    } else {
+      animate(stepperX, rect.x - parentRect.x, {
+        type: 'tween',
+        duration: 1,
+      });
+    }
+  }, [currentStepIndex]);
+
+  return (
+    <StepperWrapper {...props}>
+      <StepperTransitionIndicator key="animation" style={{x: stepperX}} />
+      <StepperContainer ref={stepperContainerRef}>
+        {Array.from(Array(numSteps).keys()).map((_, i) => (
```

Comment:
@Neo-Zhixing kind of sad there's not a better way of generating this array

Reply:
Lol yeah in any normal language we'd have Array.range(4).map(...)
or [0..4].map(...)